src/cpp/types/XRCETypes.cpp

```diff
@@ -5498,22 +5587,55 @@ size_t dds::xrce::ObjectInfo::getCdrSerializedSize(size_t current_alignment) con
 {
     size_t initial_alignment = current_alignment;
             
-    current_alignment += m_config.getCdrSerializedSize(current_alignment);
-    current_alignment += m_activity.getCdrSerializedSize(current_alignment);
+    /* Optional config. */
+    current_alignment += 1 + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+    if (m_config)
+    {
+        current_alignment += (*m_config).getCdrSerializedSize(current_alignment);
+    }
+
+    /* Optional activity. */
+    current_alignment += 1 + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+    if (m_activity)
+    {
+        current_alignment += (*m_activity).getCdrSerializedSize(current_alignment);
+    }
 
     return current_alignment - initial_alignment;
 }
 
 void dds::xrce::ObjectInfo::serialize(eprosima::fastcdr::Cdr &scdr) const
 {
-    scdr << m_config;
-    scdr << m_activity;
+    scdr << bool(m_config);
+    if (m_config)
+    {
+        scdr << (*m_config);
+    }
+    scdr << bool(m_activity);
+    if (m_activity)
+    {
+        scdr << (*m_activity);
+    }
 }
 
 void dds::xrce::ObjectInfo::deserialize(eprosima::fastcdr::Cdr &dcdr)
 {
-    dcdr >> m_config;
-    dcdr >> m_activity;
+    bool m_config_flag;
+    dcdr >> m_config_flag;
+    if (m_config_flag)
+    {
+        ObjectVariant temp_config;
+        dcdr >> temp_config;
+        m_config = temp_config;
+    }
+    bool m_activity_flag;
```

Comment:
Could reuse the same bool from `m_config_flag`.

Reply:
Grouped!